substrafl/compute_plan_builder.py

```diff
@@ -0,0 +1,77 @@
+import abc
+from pathlib import Path
+from typing import Any
+from typing import List
+from typing import Optional
+
+from substrafl.evaluation_strategy import EvaluationStrategy
+from substrafl.nodes.aggregation_node import AggregationNode
+from substrafl.nodes.train_data_node import TrainDataNode
+
+
+class ComputePlanBuilder(abc.ABC):
+    """Base compute plan builder to be inherited from SubstraFL compute plans."""
+
+    def __init__(self, *args, **kwargs):
+        """
+        All child class arguments need to be passed to it through its ``args`` and ``kwargs``
+        in order to use them when instantiating it as a RemoteStruct in each process.
+
+        Example:
+
+            .. code-block:: python
+
+                class MyComputePlan(ComputePlanBuilder):
+                    def __init__(self, custom_arg, my_custom_kwargs="value"):
+                        super().__init__(custom_arg, my_custom_kwargs=my_custom_kwargs)
+        """
+        self.args = args
+        self.kwargs = kwargs
+
+    @abc.abstractmethod
+    def build_compute_plan(
+        self,
+        train_data_nodes: Optional[List[TrainDataNode]],
+        aggregation_node: Optional[List[AggregationNode]],
+        evaluation_strategy: Optional[EvaluationStrategy],
+        num_rounds: Optional[int],
+        clean_models: Optional[bool] = True,
+    ) -> Any:
```

Comment:
Does it really returns Any? 
There is no docstring for the return, is it None?

Reply:
Nice catch... It returns None indeed